uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemLock.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.uberfire.java.nio.fs.jgit;
+
+import java.io.File;
+import java.io.RandomAccessFile;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileLock;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.uberfire.java.nio.fs.jgit.util.Git;
+
+public class JGitFileSystemLock {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JGitFileSystemLock.class);
+
+    ReentrantLock lock = new ReentrantLock(true);
+    private FileLock physicalLock;
+    private java.nio.file.Path lockFile;
+    private FileChannel fileChannel;
+    private long lastAccess;
+    private long lastAccessThreshold;
+
+    public JGitFileSystemLock(Git git, long lastAccessThreshold) {
```

Comment:
There is no indication about the expected unit to be used for the last access threshold. At minimum I'd mention the unit in the variable name (`lastAccessThresholdMiliseconds`) to avoid confusion. Alternatively you could use [TimeUnit](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/TimeUnit.html).